x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -339,7 +339,7 @@ public Sequence visitSequence(SequenceContext ctx) {
         }
 
         if (queries.stream().allMatch(KeyedFilter::isMissingEventFilter)) {
-            throw new IllegalStateException("A sequence requires at least one positive event query; found none");
+            throw new ParsingException("A sequence requires at least one positive event query; found none");
```

Comment:
super nitpick: I think parsing went fine, so maybe verification exception would be more appropriate?

Reply:
The `ParsingException` wouldn't have been 1st choice, but that's used above, so I reused it for consistency.